driver/kthvalue_driver.hpp

```diff
@@ -0,0 +1,388 @@
+        std::cerr << "Error copying (out) to GPU, size: " << output_dev->GetSize() << std::endl;
+
+    return miopenStatusSuccess;
+}
+
+template <typename TIO>
+int KthvalueDriver<TIO>::RunForwardGPU()
+{
+    float kernel_total_time = 0;
+    float kernel_first_time = 0;
+
+    Timer t;
+    START_TIME
+
+    for(int i = 0; i < inflags.GetValueInt("iter"); i++)
+    {
+        miopenKthvalueForward(GetHandle(),
+                              inputDesc,
+                              input_dev->GetMem(),
+                              outputDesc,
+                              output_dev->GetMem(),
+                              indicesDesc,
+                              (size_t*)indices_dev->GetMem(),
+                              k,
+                              dim,
+                              keepDim);
+        float time = 0.0;
+        miopenGetKernelTime(GetHandle(), &time);
+        kernel_total_time += time;
+        if(i == 0)
+            kernel_first_time = time;
+    }
+
+    if(inflags.GetValueInt("time") == 1)
+    {
+        STOP_TIME
+        int iter = inflags.GetValueInt("iter");
+        if(WALL_CLOCK)
+            std::cout << "Wall-clock Time Kthvalue Fwd Elapsed: " << t.gettime_ms() / iter << " ms"
+                      << std::endl;
+
+        float kernel_average_time =
+            iter > 1 ? (kernel_total_time - kernel_first_time) / (iter - 1) : kernel_first_time;
+        std::cout << "GPU Kernel Time Kthvalue Fwd Elapsed: " << kernel_average_time << " ms"
+                  << std::endl;
+    }
+
+    if(output_dev->FromGPU(GetStream(), output.data()) != 0)
+        std::cerr << "Error copying (out_dev) from GPU, size: " << output_dev->GetSize()
+                  << std::endl;
+
+    if(indices_dev->FromGPU(GetStream(), indices.data()) != 0)
+        std::cerr << "Error copying (indices_dev) from GPU, size: " << indices_dev->GetSize()
+                  << std::endl;
+
+    return miopenStatusSuccess;
+}
+
+template <typename TIO>
+int KthvalueDriver<TIO>::RunForwardCPU()
+{
+    mloKthvalueFwdRunHost<TIO>(input.data(),
+                               inputDesc,
+                               outputHost.data(),
+                               outputDesc,
+                               indicesHost.data(),
+                               indicesDesc,
+                               k,
+                               dim);
+
+    return miopenStatusSuccess;
+}
+
+template <typename TIO>
+int KthvalueDriver<TIO>::RunBackwardGPU()
+{
+    return miopenStatusSuccess;
+}
+
+template <typename TIO>
+int KthvalueDriver<TIO>::RunBackwardCPU()
+{
+    return miopenStatusSuccess;
+}
+
+template <typename TIO>
+int KthvalueDriver<TIO>::VerifyForward()
+{
+    RunForwardCPU();
+
+    double tolerance = std::numeric_limits<TIO>::epsilon() * 10;
+    auto errorOutput = miopen::rms_range(outputHost, output);
+
+    if(!std::isfinite(errorOutput) || errorOutput > tolerance)
```

Comment:
you should also check indices result from indicesHost

Reply:
same as the cmt above